scheds/rust/scx_layered/src/bpf/main.bpf.c

```diff
@@ -1200,6 +1225,56 @@ static bool keep_running(struct cpu_ctx *cctx, struct task_struct *p)
 	return false;
 }
 
+/*
+ * Helper function to consume layer DSQs when a system is under starvation.
+ */
+static __always_inline bool consume_starved(s32 cpu)
```

Comment:
I wonder whether it'd be simpler if we rather skip overconsuming layers rather than adding an extra step to consume starving layers.

Going back to the idea of having per-cpu candidate layer list, for each CPU, there should only be a handful of layers that a CPU should be running. If usage is tracked per-DSQ, which DSQ to consume can be decided in an a lot more granular fashion.